pandas/tests/test_generic.py

```diff
@@ -354,6 +354,52 @@ def test_head_tail(self):
             self._compare(o.head(-3), o.head(7))
             self._compare(o.tail(-3), o.tail(7))
 
+    def test_sample(self):
+        o = self._construct(shape=10)
+        
+        # Run for all index types. 
+        for index in [ tm.makeFloatIndex, tm.makeIntIndex,
+                       tm.makeStringIndex, tm.makeUnicodeIndex,
+                       tm.makeDateIndex, tm.makePeriodIndex ]:
+            axis = o._get_axis_name(0)
+            setattr(o,axis,index(len(getattr(o,axis))))
+
+            # Check seeding -- run 10 times. 
+            for test in range(10):
```

Comment:
OK -- Looping through index types to ensure if something changed in .take() we wouldn't get problems. Stability through seeding seemed like a good way to make sure everything was being passed well. I don't really test randomness of np.choice -- just stability of results with a seed. Last one was to make sure weights passing correctly. That overkill I take it?